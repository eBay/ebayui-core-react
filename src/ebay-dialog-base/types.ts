import { KeyboardEvent, SyntheticEvent, MouseEvent } from 'react'

export type DialogCloseEvent =
  | SyntheticEvent<HTMLElement>
  | KeyboardEvent<HTMLElement>
  | MouseEvent<Element>

export type DialogCloseEventHandler = (event: DialogCloseEvent) => void
