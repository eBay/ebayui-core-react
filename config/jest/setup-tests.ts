import '@testing-library/jest-dom'

declare var global: typeof globalThis;

if (typeof window.URL.createObjectURL === 'undefined') {
  Object.defineProperty(window.URL, 'createObjectURL', { value: () => {} });
}
