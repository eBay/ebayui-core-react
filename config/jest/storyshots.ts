import { render } from "@testing-library/react";
import initStoryshotsAddon from "@storybook/addon-storyshots";
import type { StoryshotsOptions } from "@storybook/addon-storyshots/dist/ts3.9/api/StoryshotsOptions";

const reactTestingLibrarySerializer = {
    print: (val, serialize, indent) => serialize(val.asFragment()),
    test: (val) => val && val.hasOwnProperty("container"),
};

export function initStoryshots(config: StoryshotsOptions) {
    initStoryshotsAddon({
        ...config,
        renderer: render,
        snapshotSerializers: [reactTestingLibrarySerializer],
    });
}
