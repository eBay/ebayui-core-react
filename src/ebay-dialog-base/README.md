# DialogBase

## DialogBase Usage

```react
<DialogBase
    open
    a11yClosetext = "Close DialogBase"
    onCloseBtnClick={ handleCloseBtnClick }>

        <h1>Hello World</h1>

</DialogBase>
```

## Attributes

| Name             | Type                      | Stateful | Required | Description                                                                                  |
|------------------|---------------------------|----------|----------|----------------------------------------------------------------------------------------------|
| `baseEl`         | string                    | No       | No       | ---  div' , 'span', 'aside'                                                                  |
| `open`           | Boolean                   | Yes      | No       | Whether drawer is open.                                                                      |
| `classPrefix`    | drawer or toast or dialog | No       | No       | "" (default) / modifies the base prefix for all  classes in the component                    |
| `windowClass`    | String                    | No       | No       | "" (default) / modifies the base prefix for all Skin classes in the component                |
| `header`         | String                    | No       | No       | element placeholder                                                                          |
| `footer`         | String                    | No       | No       | element placeholder                                                                          |
| `isModal`        | Boolean                   | No       | No       | Whether drawer is model.                                                                     |
| `top`            | ---                       | ---      | ---      | ---                                                                                          |
| `buttonPosition` | string                    | ---      | ---      | top or right or bottom or left                                                               |
| `a11yCloseText`  | String                    | No       | No*      | A11y text for close button and mask. Required only when close button exists.                 |
| `animated`       | Boolean                   | Yes      | No       | Renders the dialog with an animation. Note that the dialog will always be present in the DOM |

## Events

Event | Data | Description
`onCloseBtnClick` | --- | --- | --- | ---
`OnBackgroundClick` | --- | --- | --- | ---
