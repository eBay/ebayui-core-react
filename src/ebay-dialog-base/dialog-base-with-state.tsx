import React, { Children, ReactElement } from 'react'
import { RemoveScroll } from 'react-remove-scroll'
import { DialogBase, DialogBaseProps } from './components/dialogBase'
import { EbayDialogCloseButton, EbayDialogFooter, EbayDialogHeader } from './index'

export const DialogBaseWithState = ({
    isModal,
    open,
    children,
    animated = true,
    ...rest
}: DialogBaseProps<HTMLElement>): ReactElement => {
    const shouldRenderModal = isModal !== false

    const childrenArray = Children.toArray(children) as ReactElement[]
    const header = childrenArray.find((child: ReactElement) => child.type === EbayDialogHeader)
    const footer = childrenArray.find((child: ReactElement) => child.type === EbayDialogFooter)
    const closeButton = childrenArray.find((child: ReactElement) => child.type === EbayDialogCloseButton)
    const content = childrenArray.filter((child: ReactElement) =>
        ![EbayDialogHeader, EbayDialogFooter, EbayDialogCloseButton].some(c => c === child.type))

    const dialogBase = (
        <DialogBase
            {...rest}
            open={open}
            isModal={shouldRenderModal}
            header={header}
            footer={footer}
            closeButton={closeButton}
            animated={animated}
        >
            {content}
        </DialogBase>
    )

    const renderOverLay = () => shouldRenderModal ? (
        <RemoveScroll enabled={open}>
            {dialogBase}
        </RemoveScroll>
    ) : dialogBase

    return animated || open ? renderOverLay() : null
}

export default DialogBaseWithState
