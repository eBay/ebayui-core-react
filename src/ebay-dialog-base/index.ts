export { default as EbayDialogHeader } from './components/dialog-header'
export { default as EbayDialogFooter } from './components/dialog-footer'
export { default as EbayDialogCloseButton } from './components/dialog-close-button'
export { WindowType, DialogBaseProps } from './components/dialogBase'
export { default as DialogBaseWithState } from './dialog-base-with-state'
export { default as EbayDialogActions } from './components/dialog-actions'
