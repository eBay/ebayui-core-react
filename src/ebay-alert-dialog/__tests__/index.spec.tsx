import React from 'react'
import requireContext from 'node-require-context'
import { screen, fireEvent, render, RenderResult } from '@testing-library/react'
import { EbayAlertDialog } from '../index';
import { initStoryshots } from '../../../config/jest/storyshots'
import { EbayDialogHeader } from '../../ebay-dialog-base'

jest.mock('../../common/random-id', () => ({ randomId: () => 'abc123' }))

const closeSpy = jest.fn()
const openSpy = jest.fn()
const openDialog = () =>
    render(
        <EbayAlertDialog
            confirmText="Confirm"
            open
            onOpen={openSpy}
            onConfirm={closeSpy}
            a11yCloseText="Close"
        >
            <EbayDialogHeader>Heading</EbayDialogHeader>
            <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis
                nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
            </p>
            <p><a href="http://www.ebay.com">www.ebay.com</a></p>
            <input placeholder="Enter a value" />
        </EbayAlertDialog>
    )

describe('<EbayAlertDialog>', () => {
    it('should trigger onOpen when dialog appears', () => {
        openDialog()
        expect(openSpy).toBeCalled()
    })

    it('should have close button', () => {
        openDialog()
        expect(screen.getByText('Confirm')).toBeInTheDocument()
    })

    it('should trigger onClose when close button is clicked', () => {
        openDialog()
        fireEvent.click(screen.getByText('Confirm'))
        expect(closeSpy).toBeCalled()
    })
})

initStoryshots({
    config: ({ configure }) => {
        const req = requireContext('./', false, /\.stories\.tsx$/);
        return configure(req, module)
    }
});
