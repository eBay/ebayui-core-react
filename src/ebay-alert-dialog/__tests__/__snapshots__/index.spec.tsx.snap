// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-alert-dialog Default 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
  >
    <button
      aria-hidden="true"
      class="btn btn--secondary"
    >
      Open Dialog
    </button>
    <p
      aria-hidden="true"
    >
      Some outside content...
    </p>
    <div
      aria-hidden="false"
    >
      <div
        aria-hidden="false"
        aria-labelledby="dialog-title-abc123"
        arial-modal="true"
        class="alert-dialog alert-dialog--mask-fade alert-dialog--show-init"
        role="alertdialog"
      >
        <div
          aria-hidden="true"
          class="keyboard-trap-boundary"
          tabindex="0"
        />
        <div
          aria-hidden="false"
          class="alert-dialog__window alert-dialog__window--fade keyboard-trap--active"
        >
          <div
            aria-hidden="true"
            class="keyboard-trap-boundary"
            tabindex="0"
          />
          <div
            class="alert-dialog__header"
          >
            <h2
              id="dialog-title-abc123"
            >
              Heading
            </h2>
          </div>
          <div
            class="alert-dialog__main"
            id="alert-dialog-main"
          >
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
            </p>
            <p>
              <a
                href="http://www.ebay.com"
              >
                www.ebay.com
              </a>
            </p>
          </div>
          <div
            class="alert-dialog__footer"
          >
            <button
              aria-describedby="alert-dialog-main"
              class="btn alert-dialog__acknowledge btn--primary"
              id="alert-dialog-confirm"
            >
              Confirm
            </button>
          </div>
          <div
            aria-hidden="true"
            class="keyboard-trap-boundary"
            tabindex="0"
          />
        </div>
        <div
          aria-hidden="true"
          class="keyboard-trap-boundary"
          tabindex="0"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-alert-dialog With Animation 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn btn--secondary"
    >
      Open Dialog
    </button>
    <p>
      Some outside content...
    </p>
    <div>
      <div
        aria-labelledby="dialog-title-abc123"
        arial-modal="true"
        class="alert-dialog alert-dialog--mask-fade"
        hidden=""
        role="alertdialog"
      >
        <div
          class="alert-dialog__window alert-dialog__window alert-dialog__window--fade"
        >
          <div
            class="alert-dialog__header"
          >
            <h2
              id="dialog-title-abc123"
            >
              Heading
            </h2>
          </div>
          <div
            class="alert-dialog__main"
            id="alert-dialog-main"
          >
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
            </p>
            <p>
              <a
                href="http://www.ebay.com"
              >
                www.ebay.com
              </a>
            </p>
          </div>
          <div
            class="alert-dialog__footer"
          >
            <button
              aria-describedby="alert-dialog-main"
              class="btn alert-dialog__acknowledge btn--primary"
              id="alert-dialog-confirm"
            >
              Confirm
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
